src/test/java/com/hotels/hcommon/hive/metastore/client/closeable/CloseableMetaStoreClientInvocationHandlerTest.java

```diff
@@ -68,22 +71,66 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
     verify(compatibility).getTable(eq(dbName), eq(tableName));
   }
 
-  @Test(expected = RuntimeException.class)
+  @Test(expected = NoSuchObjectException.class)
   public void dontInvokeCompatibilityWhenExceptionOtherThanTApplicationExceptionIsThrown() throws Throwable {
     Class<?> clazz = Class.forName(IMetaStoreClient.class.getName());
     Method method = clazz.getMethod("getTable", String.class, String.class);
     String dbName = "db";
     String tableName = "table";
 
+    IMetaStoreClient exceptionThrowingClient = Mockito.mock(IMetaStoreClient.class, new Answer<Void>() {
+      @Override
+      public Void answer(InvocationOnMock invocation) throws Throwable {
+        throw new NoSuchObjectException();
+      }
+    });
 
-    IMetaStoreClient exceptionThrowingClient = Mockito.mock(IMetaStoreClient.class, new Answer() {
+    invocationHandler = new CloseableMetaStoreClientInvocationHandler(exceptionThrowingClient, compatibility);
+    invocationHandler.invoke(null, method, new String[] { dbName, tableName });
+  }
+
+  @Test(expected = NoSuchObjectException.class)
+  public void rethrowCompatibilityClassException() throws Throwable {
+    Class<?> clazz = Class.forName(IMetaStoreClient.class.getName());
+    Method method = clazz.getMethod("getTable", String.class, String.class);
+    String dbName = "db";
+    String tableName = "table";
+
+    IMetaStoreClient exceptionThrowingClient = Mockito.mock(IMetaStoreClient.class, new Answer<Void>() {
       @Override
-      public Object answer(InvocationOnMock invocation) throws Throwable {
-        throw new RuntimeException();
```

Comment:
Shouldn't we still test what happens if something unexpected like a `RuntimeException` is thrown?

Reply:
yes adding that now